src/map/packets/char_update.cpp

```diff
@@ -90,8 +92,20 @@ CCharUpdatePacket::CCharUpdatePacket(CCharEntity* PChar)
     
     ref<uint8>(0x36) = flag;
 
-    if (!PChar->isDead() || PChar->m_DeathCounter == 0) //prevent this packet from resetting the homepoint timer after tractor
+    //This contains the amount of time remaining before the player is forced back to homepoint while dead
+    if (PChar->PAI->IsCurrentState<CDeathState>())
+    {
+        //Update the player death counter and timestamp and send the amount of time remaining
+        uint32 currentTime = (uint32)time(nullptr);
+        PChar->m_DeathCounter += (currentTime - PChar->m_DeathTimestamp);
+        PChar->m_DeathTimestamp = currentTime;
```

Comment:
real lost here - why is a packet updating member info? why is the death timestamp being constantly overwritten with current time?
Maybe it's because the way we're currently dealing with death timers (two member variables) is real wrong, but I guess this presents an opportunity to fix it

Reply:
You bring up a good point. It is wrong for the packet to be updating member info. This could be changed to the following so it is not updating member info

```c++
uint32 currentTime = (uint32)time(nullptr);
int32 counter = PChar->m_DeathCounter + (currentTime - PChar->m_DeathTimestamp);
ref<uint32>(0x3C) = 0x0003A020 - (60 * counter);
```

I can look into consolidating these two variables into a single one though.